examples/userapi/05_conditional_dimension.ipynb

```diff
@@ -0,0 +1,882 @@
+      "  /* Flush denormal numbers to zero in hardware */\n",
+      "  _MM_SET_DENORMALS_ZERO_MODE(_MM_DENORMALS_ZERO_ON);\n",
+      "  _MM_SET_FLUSH_ZERO_MODE(_MM_FLUSH_ZERO_ON);\n",
+      "  struct timeval start_section0, end_section0;\n",
+      "  gettimeofday(&start_section0, NULL);\n",
+      "  /* Begin section0 */\n",
+      "  #pragma omp parallel num_threads(nthreads)\n",
+      "  {\n",
+      "    #pragma omp for collapse(1) schedule(static,1)\n",
+      "    for (int x = x_m; x <= x_M; x += 1)\n",
+      "    {\n",
+      "      #pragma omp simd aligned(f,g:32)\n",
+      "      for (int y = y_m; y <= y_M; y += 1)\n",
+      "      {\n",
+      "        if (y < 5 && g[x + 1][y + 1] != 0)\n",
+      "        {\n",
+      "          f[x + 1][y + 1] = f[x + 1][y + 1] + g[x + 1][y + 1];\n",
+      "        }\n",
+      "      }\n",
+      "    }\n",
+      "  }\n",
+      "  /* End section0 */\n",
+      "  gettimeofday(&end_section0, NULL);\n",
+      "  timers->section0 += (double)(end_section0.tv_sec-start_section0.tv_sec)+(double)(end_section0.tv_usec-start_section0.tv_usec)/1000000;\n",
+      "  return 0;\n",
+      "}\n",
+      "\n"
+     ]
+    },
+    {
+     "data": {
+      "text/plain": [
+       "Data([[2., 0., 0., 0., 0., 0., 0., 0., 0., 0.],\n",
+       "      [0., 2., 0., 0., 0., 0., 0., 0., 0., 0.],\n",
+       "      [0., 0., 2., 0., 0., 0., 0., 0., 0., 0.],\n",
+       "      [0., 0., 0., 2., 0., 0., 0., 0., 0., 0.],\n",
+       "      [0., 0., 0., 0., 2., 0., 0., 0., 0., 0.],\n",
+       "      [0., 0., 0., 0., 0., 1., 0., 0., 0., 0.],\n",
+       "      [0., 0., 0., 0., 0., 0., 1., 0., 0., 0.],\n",
+       "      [0., 0., 0., 0., 0., 0., 0., 1., 0., 0.],\n",
+       "      [0., 0., 0., 0., 0., 0., 0., 0., 1., 0.],\n",
+       "      [0., 0., 0., 0., 0., 0., 0., 0., 0., 1.]], dtype=float32)"
+      ]
+     },
+     "execution_count": 7,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "#NBVAL_IGNORE_OUTPUT\n",
+    "\n",
+    "from sympy import And\n",
+    "from devito.types import Ne, Lt\n",
+    "\n",
+    "f.data[:,:] = np.eye(10)\n",
+    "\n",
+    "\n",
+    "g = Function(name='g', grid=grid)\n",
+    "g.data[:,:] = np.eye(10)\n",
+    "\n",
+    "ci = ConditionalDimension(name='ci', parent=y, condition=And(Ne(g, 0), Lt(y, 5)))\n",
+    "\n",
+    "op = Operator(Eq(f, f + g, implicit_dims=ci))\n",
+    "op.apply()\n",
+    "\n",
+    "print(op.ccode)\n",
+    "\n",
+    "assert (np.count_nonzero(f.data - np.diag(np.diagonal(f.data)))==0)\n",
+    "assert (np.count_nonzero(f.data) == 10)\n",
+    "assert np.all(f.data[np.nonzero(f.data[:5,:5])] == 2)\n",
+    "assert np.all(f.data[5:,5:] == np.eye(5))\n",
+    "\n",
+    "f.data"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "You can see that now `f` has been updated only for the first five rows with the `f+g` expression."
```

Comment:
Your conditional is on `y` which would correspond to the columns of `f.data` so you should be referring to column's instead of rows in the text right?

Reply:
Yes, correct.